src/sdl2_compat.c

```diff
@@ -2491,63 +2491,46 @@ SDL_GetTicks64(void)
     return SDL3_GetTicks();
 }
 
-static SDL_bool
-SysWMInfo3to2(SDL_SysWMinfo *wminfo3, SDL2_SysWMinfo *wminfo2)
-{
-    switch (wminfo3->subsystem) {
-#if defined(SDL_ENABLE_SYSWM_ANDROID)
-    case SDL_SYSWM_ANDROID:
-        wminfo2->subsystem = SDL2_SYSWM_ANDROID;
-        wminfo2->info.android.window = wminfo3->info.android.window;
-        wminfo2->info.android.surface = wminfo3->info.android.surface;
-        break;
-#endif
-#if defined(SDL_ENABLE_SYSWM_COCOA)
-    case SDL_SYSWM_COCOA:
-        wminfo2->subsystem = SDL2_SYSWM_COCOA;
-        wminfo2->info.cocoa.window = wminfo3->info.cocoa.window;
-        break;
-#endif
-#if defined(SDL_ENABLE_SYSWM_DIRECTFB)
-    case SDL_SYSWM_DIRECTFB:
-        wminfo2->info.dfb.dfb = wminfo3->info.dfb.dfb;
-        wminfo2->info.dfb.window = wminfo3->info.dfb.window;
-        wminfo2->info.dfb.surface = wminfo3->info.dfb.surface;
-        break;
-#endif
-#if defined(SDL_ENABLE_SYSWM_KMSDRM)
-    case SDL_SYSWM_KMSDRM:
-        wminfo2->subsystem = SDL2_SYSWM_KMSDRM;
-        wminfo2->info.kmsdrm.dev_index = wminfo3->info.kmsdrm.dev_index;
-        wminfo2->info.kmsdrm.drm_fd = wminfo3->info.kmsdrm.drm_fd;
-        wminfo2->info.kmsdrm.gbm_dev = wminfo3->info.kmsdrm.gbm_dev;
-        break;
-#endif
-#if defined(SDL_ENABLE_SYSWM_OS2)
-    case SDL_SYSWM_OS2:
-        wminfo2->subsystem = SDL2_SYSWM_OS2;
-        wminfo2->info.os2.hwnd = wminfo3->info.os2.hwnd;
-        wminfo2->info.os2.hwndFrame = wminfo3->info.os2.hwndFrame;
-        break;
-#endif
-#if defined(SDL_ENABLE_SYSWM_UIKIT)
-    case SDL_SYSWM_COCOA:
-        wminfo2->subsystem = SDL2_SYSWM_UIKIT;
-        wminfo2->info.uikit.window = wminfo3->info.uikit.window;
-        break;
-#endif
-#if defined(SDL_ENABLE_SYSWM_VIVANTE)
-    case SDL_SYSWM_VIVANTE:
-        wminfo2->subsystem = SDL2_SYSWM_VIVANTE;
-        wminfo2->info.vivante.display = wminfo3->info.vivante.display;
-        wminfo2->info.vivante.window = wminfo3->info.vivante.window;
-        break;
-#endif
-#if defined(SDL_ENABLE_SYSWM_WAYLAND)
-    case SDL_SYSWM_WAYLAND: {
-        Uint32 version2 = SDL_VERSIONNUM((Uint32)wminfo2->version.major,
-                                         (Uint32)wminfo2->version.minor,
-                                         (Uint32)wminfo2->version.patch);
+DECLSPEC SDL_bool SDLCALL SDL_GetWindowWMInfo(SDL_Window *window, SDL_SysWMinfo *wminfo)
+{
+    const char *driver = SDL3_GetCurrentVideoDriver();
+    SDL_PropertiesID props = SDL3_GetWindowProperties(window);
+
+    if (!wminfo) {
+        return SDL_FALSE;
+    }
+
+    if (!driver) {
+        SDL_SetError("No video driver initialized");
+        return SDL_FALSE;
+    }
+
+    if (SDL_strcmp(driver, "Android") == 0) {
+        wminfo->subsystem = SDL2_SYSWM_ANDROID;
+        wminfo->info.android.window = (void *)SDL3_GetProperty(props, "SDL.window.android.window");
```

Comment:
The cast to (void *) isn't needed, here or below.

Reply:
Fixed